core_js/popup.js

```diff
@@ -62,6 +62,40 @@ function changeStatistics()
     elTotal.textContent = totalCounter.toLocaleString();
 }
 
+/**
+* Set the whitelist button text
+*/
+function setWhitelistText()
+{
+    let element = document.getElementById('whitelist_btn');
+    let currentSite, siteFound;
+    browser.tabs.query({active: true, currentWindow: true}, function(tabs) {
+        currentSite = tabs[0].url;
+    });
+
+    browser.runtime.sendMessage({
+        function: "getData",
+        params: ['whitelist']
+    }).then((data) => {
+        data.response.forEach(site => {
+            if (currentSite.indexOf(site) != -1) {
+                siteFound = true
+            }
+        });
+        if (!siteFound) {
+            if (data.response.length != 0) {
```

Comment:
Remove this If. It causes the button to retain the "btn-danger" class when a page is added and immediately removed.